### ISSUE TITLE
Gamelift realtime client stops receiving messages in Unity 2019

### ISSUE THREAD

**Issue by danilonobren** [aws/aws-sdk-net#1429]
User "MidgetJake" on stackoverflow asked the following and it doesn't seem like he received any response:

> "Amazon GameLift realtime client stops receiving messages after being accepted"
> https://stackoverflow.com/questions/57628984/amazon-gamelift-realtime-client-stops-receiving-messages-after-being-accepted

I followed the same steps as him (accidentally) in Unity 2019 and got the same results.

Unity seems to do nothing when the packages are received (confirmed by Wireshark) even though _Client_ gets confirmation that _Player_ got connected/accepted and is sending messages to _Server_.

#### Client sends message with spacebar

```CSharp
if (client != null && Input.GetKeyDown(KeyCode.Space)) {
            RTMessage message = client.NewMessage(OP_CODE_CUSTOM_OP)
                .WithDeliveryIntent(DeliveryIntent.Reliable)
                .WithTargetPlayer(Constants.PLAYER_ID_SERVER)
                .WithPayload(StringToBytes("TEST"));

            ChatLog.CreateLog("[client] Send Message: 'TEST'");
            client.SendMessage(message);

            if (!client.ConnectedAndReady) {
                Debug.Log("Not connected.");
                ChatLog.CreateLog("Not connected.");
            }
}
```

#### Receiving packages
```CSharp
private void OnDataReceived(object sender, DataReceivedEventArgs e) {
        string data = System.Text.Encoding.Default.GetString(e.Data);
        Debug.Log($"[server-sent] OnDataReceived - Sender: {e.Sender} OpCode: {e.OpCode} data: {data}");

        switch (e.OpCode) {
            case Constants.LOGIN_RESPONSE_OP_CODE:
                peerID = e.Sender;
                Debug.Log($"[server] peer ID : {peerID}");

                break;
            case OP_CODE_PLAYER_CONNECT:
                peerID = e.Sender;
                Debug.Log($"[server] {peerID}: Connected");

                break;
            case OP_CODE_PLAYER_ACCEPTED:
                peerID = e.Sender;
                Debug.Log($"[server] {peerID}: Accepted");

                break;
            case OP_CODE_PLAYER_DISCONNECT:
                peerID = e.Sender;
                Debug.Log($"[server] {peerID}: Disconnected");

                break;
            case OP_CODE_CUSTOM_OP_REPLY:
                peerID = e.Sender;
                Debug.Log($"[client] {peerID}: Hello :)");
                ChatLog.CreateLog($"[client] {peerID}: Hello :)");

                break;
        }
}
```

It never receives a "Hello" back :(

## Your Environment
**Unity 2019.1.0f2**
- Runtime Configuration: .NET 4.x Equivalent
- Scripting Backend: Mono
- Api Compability Level: .NET 4.x

**AWSSDK.GameLift 3.3.103.16**
**AWSSDK.Core 3.3.103.48**
- Targeted .NET platform: 4.5

**Windows 10**
**Visual Studio 2019**

**Comment by klaytaybai** [comment]
Unfortunately I don't think there is a lot that our team is going to be able to do to help you with this.  The GameLift team manages the [Realtime Client SDK](https://aws.amazon.com/gamelift/getting-started/) themselves.  I'll try to get somebody from their team to address this issue.  If you have Premium Support, that might be an effective channel for communicating with the GameLift team.

**Comment by danilonobren** [comment]
> Unfortunately I don't think there is a lot that our team is going to be able to do to help you with this. The GameLift team manages the [Realtime Client SDK](https://aws.amazon.com/gamelift/getting-started/) themselves. I'll try to get somebody from their team to address this issue. If you have Premium Support, that might be an effective channel for communicating with the GameLift team.

I see... I'll seek for some alternatives... 
Thank you!

**Comment by alanmur** [comment]
I'm thinking this may be a simple bug in your code. What happens if you have a case in your switch statement for default. i.e.
`default:
              Console.WriteLine("Another sort of message arrived, maybe the one you are expecting");
              break;`
I bet that would receive your message. If so, you should verify that the OPCODE value you are sending is the same one as the one you are testing for in your switch statement.

If this does not work, I will repro your problem and follow up with you on email.

Al Murray :)
Solutions Architect, Amazon Game Tech

**Comment by danilonobren** [comment]
> I'm thinking this may be a simple bug in your code. What happens if you have a case in your switch statement for default. i.e.
> `default: Console.WriteLine("Another sort of message arrived, maybe the one you are expecting"); break;`
> I bet that would receive your message. If so, you should verify that the OPCODE value you are sending is the same one as the one you are testing for in your switch statement.
> 
> The way it looks now, OP_CODE_CUSTOM_OP would need to be a case in the switch statement, not OP_CODE_CUSTOM_OP_REPLY which probably has a different value.
> 
> If this does not work, I will repro your problem and follow up with you on email.
> 
> Al Murray :)
> Solutions Architect, Amazon Game Tech

Thank you for your response @alanmur 

There's a log before the switch case that never gets triggered.

`Debug.Log($"[server-sent] OnDataReceived - Sender: {e.Sender} OpCode: {e.OpCode} data: {data}");`

In our code, server should be getting any message with OP_CODE_CUSTOM_OP and sending it back with OP_CODE_CUSTOM_OP_REPLY.

Here's the function on server.

```
function onMessage(gameMessage) {
    switch (gameMessage.opCode) {
        case OP_CODE_CUSTOM_OP:
            logger.info("CUSTOM_OP");
            const msg = session.newTextGameMessage(OP_CODE_CUSTOM_OP_REPLY, session.getServerId(), gameMessage.payload);
            session.getPlayers().forEach((p) => {
                session.sendReliableMessage(msg, p.peerId);
            });
            break;
    }
}
```

**Comment by alanmur** [comment]
OK I will look further and try and repro the problem on my end.

**Comment by NGL321** [comment]
Hey @danilonobren,

Quick heads-up, [as of SDK v3.5](https://aws.amazon.com/blogs/developer/aws-sdk-for-net-v3-5-now-generally-available/) we no longer separate Unity support from regular .NET support, if you are having trouble with this transition, please make sure you are [targeting ^.NET 4.0 or ^.NET Standard 2.0](https://docs.unity3d.com/2020.1/Documentation/Manual/dotnetProfileSupport.html) using Unity 2018.0 or later. This means that you can have full access to AWS services when using the .NET SDK with Unity, but if you would still like to use the legacy SDK for Unity, you can reference it from our [archive repo](https://github.com/aws/aws-sdk-unity-net).

I see that this was likely a Gamelift issue though, @alanmur were you able to reproduce this problem? If so, does it persist with the new version of the SDK?

Please let me know and I will adjust this issue accordingly.

😸 😷

**Comment by alanmur** [comment]
@NGL321 I am not able to repro the problem. I used this project https://github.com/aws-samples/megafrograce-gamelift-realtime-servers-sample to repro. The log looks like the server in this case is successfully sending data to the client and the client is catching it.

Server Log:
```
17 Sep 2020 17:50:06,677 [INFO] (MegaFrogRaceServer.js) 194: [app] onPlayerAccepted: player.peerId = 2
17 Sep 2020 17:50:06,677 [INFO] (MegaFrogRaceServer.js) 199: [app] onPlayerAccepted: new contents of players array = 1,2
17 Sep 2020 17:50:06,677 [INFO] (MegaFrogRaceServer.js) 202: [app] onPlayerAccepted: logical ID = 1
17 Sep 2020 17:50:06,677 [INFO] (MegaFrogRaceServer.js) 205: [app] onPlayerAccepted: logicalPlayerIDs array = [object Object]
17 Sep 2020 17:50:06,677 [INFO] (MegaFrogRaceServer.js) 48: [app] SendStringToClient: peerIds = 2 opCode = 100 stringToSend = 1
17 Sep 2020 17:50:06,677 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 2
17 Sep 2020 17:50:06,677 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 100, targetPeerId: 2, targetGroupId: 0
17 Sep 2020 17:50:06,678 [INFO] (index.js) 430: Sending login result to player: 2
17 Sep 2020 17:50:06,714 [INFO] (index.js) 449: Sending server ack to player 2. iteration=0
17 Sep 2020 17:50:06,751 [INFO] (index.js) 282: Player 2 fully setup over ws + udp
17 Sep 2020 17:50:06,777 [INFO] (index.js) 346: Received custom game message from sender: 2 with opCode: 200, targetPeerId: -1, targetGroupId: 0
17 Sep 2020 17:50:06,777 [INFO] (MegaFrogRaceServer.js) 142: [app] onMessage(gameMessage):
17 Sep 2020 17:50:06,777 [INFO] (MegaFrogRaceServer.js) 143: { opCode: 200,
  sender: 2,
  payload: Uint8Array [],
  reliable: false }
17 Sep 2020 17:50:06,777 [INFO] (MegaFrogRaceServer.js) 48: [app] SendStringToClient: peerIds = 1,2 opCode = 101 stringToSend = 4
17 Sep 2020 17:50:06,778 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 1
17 Sep 2020 17:50:06,778 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 101, targetPeerId: 1, targetGroupId: 0
17 Sep 2020 17:50:06,778 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 2
17 Sep 2020 17:50:06,778 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 101, targetPeerId: 2, targetGroupId: 0
17 Sep 2020 17:50:06,814 [INFO] (index.js) 440: Player 2 is now fully connected for tcp and udp
17 Sep 2020 17:50:10,681 [INFO] (index.js) 346: Received custom game message from sender: 1 with opCode: 201, targetPeerId: -1, targetGroupId: 0
17 Sep 2020 17:50:10,681 [INFO] (MegaFrogRaceServer.js) 142: [app] onMessage(gameMessage):
17 Sep 2020 17:50:10,681 [INFO] (MegaFrogRaceServer.js) 143: { opCode: 201,
  sender: 1,
  payload: Uint8Array [],
  reliable: false }
17 Sep 2020 17:50:10,682 [INFO] (MegaFrogRaceServer.js) 48: [app] SendStringToClient: peerIds = 1,2 opCode = 102 stringToSend = 0:0.2
17 Sep 2020 17:50:10,682 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 1
17 Sep 2020 17:50:10,682 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 102, targetPeerId: 1, targetGroupId: 0
17 Sep 2020 17:50:10,682 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 2
17 Sep 2020 17:50:10,683 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 102, targetPeerId: 2, targetGroupId: 0
17 Sep 2020 17:50:13,684 [INFO] (index.js) 346: Received custom game message from sender: 2 with opCode: 201, targetPeerId: -1, targetGroupId: 0
17 Sep 2020 17:50:13,684 [INFO] (MegaFrogRaceServer.js) 142: [app] onMessage(gameMessage):
17 Sep 2020 17:50:13,684 [INFO] (MegaFrogRaceServer.js) 143: { opCode: 201,
  sender: 2,
  payload: Uint8Array [],
  reliable: false }
17 Sep 2020 17:50:13,684 [INFO] (MegaFrogRaceServer.js) 48: [app] SendStringToClient: peerIds = 1,2 opCode = 102 stringToSend = 1:0.2
17 Sep 2020 17:50:13,684 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 1
17 Sep 2020 17:50:13,685 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 102, targetPeerId: 1, targetGroupId: 0
17 Sep 2020 17:50:13,685 [INFO] (MegaFrogRaceServer.js) 54: [app] SendStringToClient: sendMessageT [object Object] 2
17 Sep 2020 17:50:13,685 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 102, targetPeerId: 2, targetGroupId: 0
17 Sep 2020 17:50:16,653 [INFO] (index.js) 346: Received custom game message from sender: 1 with opCode: 201, targetPeerId: -1, targetGroupId: 0
17 Sep 2020 17:50:16,654 [INFO] (MegaFrogRaceServer.js) 142: [app] onMessage(gameMessage):
17 Sep 2020 17:50:16,654 [INFO] (MegaFrogRaceServer.js) 143: { opCode: 201,
  sender: 1,
  payload: Uint8Array [],
  reliable: false }

```

Client Logs - both clients writing to same log :(
```
[INFO] Sending packet with size 19, opcode: 200, content: { "opCode": 200, "sequenceNumber": 2, "targetPlayer": -1, "sender": 2 }
[server-sent] OnDataReceived - Sender: -1 OpCode: 101 data: 4
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 1:0.2
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 1:0.4
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 1:0.6000000000000001
SendingEvent: SCENE_READY_OP_CODE
[INFO] Sending packet with size 19, opcode: 201, content: { "opCode": 201, "sequenceNumber": 3, "targetPlayer": -1, "sender": 1 }
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 0:0.2
SendingEvent: SCENE_READY_OP_CODE
[INFO] Sending packet with size 19, opcode: 201, content: { "opCode": 201, "sequenceNumber": 4, "targetPlayer": -1, "sender": 1 }
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 0:0.4
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 1:0.8
[server-sent] OnDataReceived - Sender: -1 OpCode: 102 data: 1:1
[server-sent] OnDataReceived - Sender: -1 OpCode: 103 data: 1:0
```

Can the OP get this sample to run?

**Comment by jakedv24** [comment]
@NGL321 I also followed the MegaFrogRace example closely in my implementation of a realtime GameLift client/server. https://github.com/aws-samples/megafrograce-gamelift-realtime-servers-sample

I believe I've been running into the same or a similar issue. From what I have gathered from both the server and client logs, I am able to see the players connect and be accepted on the server, but the client will then hang when calling connect:

```
        print($"[client] Attempting to connect to server ip: {ipAddr} TCP port: {port} Player Session ID: {tokenUID}");
        // TODO - getting stuck here when attempting connection.
        // Server logs show connection
        _client.Connect(string.IsNullOrEmpty(ipAddr) ? DEFAULT_ENDPOINT : ipAddr, port, UDPListenPort, token);
```

I can verify that the ip address and ports are all correct to what the AWS Console shows for the game session, and that the server logs show a connected and accepted player...

```
04 Jan 2021 23:31:33,370 [INFO] (index.js) 176: [app] onStartGameSession
04 Jan 2021 23:31:33,371 [INFO] (gamelift.js) 39: GameLiftServerAPI completed StartGameSession
04 Jan 2021 23:31:34,284 [INFO] (index.js) 469: Player 1 connected from 69.5.132.129
04 Jan 2021 23:31:34,284 [INFO] (ws.js) 139: Player 1 connecting from ::ffff:69.5.132.129:57650
04 Jan 2021 23:31:34,434 [INFO] (index.js) 202: Message received from player 1
04 Jan 2021 23:31:34,436 [INFO] (index.js) 210: Packet (including both internal messages and custom game messages) received from player: 1
04 Jan 2021 23:31:34,436 [INFO] (index.js) 378: Handling login request from player: 1
04 Jan 2021 23:31:34,436 [INFO] (index.js) 381: Player 1 logging in
04 Jan 2021 23:31:34,436 [INFO] (index.js) 388: Player: 1 is connecting to player session: psess-f2b61852-3cc0-48b9-ab15-bddd220aff0e
04 Jan 2021 23:31:34,437 [INFO] (gamelift.js) 83: Calling GameLiftServerAPI.AcceptPlayerSession with player session id: psess-f2b61852-3cc0-48b9-ab15-bddd220aff0e
04 Jan 2021 23:31:34,458 [INFO] (gamelift.js) 88: GameLiftServerAPI.AcceptPlayerSession succeeded with player session id: psess-f2b61852-3cc0-48b9-ab15-bddd220aff0e
04 Jan 2021 23:31:34,458 [INFO] (index.js) 395: Successfully connected player: 1 to player session:  psess-f2b61852-3cc0-48b9-ab15-bddd220aff0e
04 Jan 2021 23:31:34,459 [INFO] (index.js) 118: [app] onPlayerConnect: undefined
04 Jan 2021 23:31:34,459 [INFO] (index.js) 137: [app] onPlayerAccepted: player.peerId = 1
04 Jan 2021 23:31:34,459 [INFO] (index.js) 144: [app] onPlayerAccepted: new contents of players array = 1
04 Jan 2021 23:31:34,459 [INFO] (index.js) 147: [app] onPlayerAccepted: logical ID = 0
04 Jan 2021 23:31:34,459 [INFO] (index.js) 151: [app] onPlayerAccepted: logicalPlayerIDs array = [object Object]
04 Jan 2021 23:31:34,459 [INFO] (index.js) 29: [app] SendStringToClient: peerIds = 1 opCode = 1 stringToSend = 0
04 Jan 2021 23:31:34,459 [INFO] (index.js) 40: [app] SendStringToClient: sendMessageT [object Object] 1
04 Jan 2021 23:31:34,460 [INFO] (index.js) 346: Received custom game message from sender: -1 with opCode: 1, targetPeerId: 1, targetGroupId: 0
04 Jan 2021 23:31:34,461 [INFO] (index.js) 430: Sending login result to player: 1
04 Jan 2021 23:31:34,596 [INFO] (index.js) 449: Sending server ack to player 1. iteration=0
04 Jan 2021 23:31:34,689 [INFO] (index.js) 282: Player 1 fully setup over ws + udp
04 Jan 2021 23:31:34,697 [INFO] (index.js) 440: Player 1 is now fully connected for tcp and udp
04 Jan 2021 23:32:06,383 [INFO] (gamelift.js) 71: (Heartbeat) GameLiftServerAPI initiating OnHealthCheck
```

From my testing, this has the effect of blocking messages from the server, as well as any user input. Since user input is blocked, I figured it must be running and blocking on the main thread. Looking at the example code that I brought in from the MegaFrogRace example, it should be running everything in a non-blocking coroutine. I'm newer to the Unity and C# space and I'm unsure how coroutines work completely. But from what I've researched, they should relinquish control of the main thread periodically for other tasks to run.

When I initialize the server connection, I already have known game session info to connect to (I make the call earlier to grab the session info), so I **don't** need to call the Lambda used in the example with:

```
        client.InvokeAsync(request,
            (response) =>
            {
                if(response.Exception == null)
                {
                    if(response.Response.StatusCode == 200)
                    {
                        var payload = Encoding.ASCII.GetString(response.Response.Payload.ToArray()) + "\n";
                        var playerSessionObj = JsonUtility.FromJson<PlayerSessionObject>(payload);

                        if (playerSessionObj.FleetId == null)
                        {
                            Debug.Log($"Error in Lambda: {payload}");
                        }
                        else
                        {
                            QForMainThread(ActionConnectToServer, playerSessionObj.IpAddress, Int32.Parse(playerSessionObj.Port), playerSessionObj.PlayerSessionId);
                        }
                    }
                }
                else
                {
                    Debug.LogError(response.Exception);
                }
            });
```

Would this invoke async have any affects on the thread that gets executed when emptying the queue of coroutines? That is all that I can see differing from the original MegaFrogRace example.

**Comment by jakedv24** [comment]
@NGL321 @alanmur 

I was able to get the MegaFrogRace up and running correctly on Android devices. However, for our use we require `il2cpp` as the scripting backend. I can confirm that builds will hang on connect when using `il2cpp` but will not hang when using `Mono` as the scripting backend.

**Note: Only tested on Android builds**

I isolated any other changes in my testing, swapping between `il2cpp` and `Mono` when building would fix / cause the hanging issue with `_client.connect(...);`

**Comment by hankditton** [comment]
I'm experiencing the same issue described in the OP:

- Unity 2020.3
- Windows 10
- Latest version of Realtime Client SDK
- Client seems to be connecting successfully to server, and can send messages to the server (verified server logs show messages being received). 
- Client is not getting any messages from the server (or at least the OnDataEvent callback is never being triggered).

The weird thing is this was working at first, but now is no longer working (and I don't think I changed anything?)

Sample server log that isn't working:
`09 Jun 2021 04:08:20,709 [INFO] (index.js) 471: Player 1 connected from <ip>
09 Jun 2021 04:08:20,709 [INFO] (ws.js) 139: Player 1 connecting from ::ffff:<ip>
09 Jun 2021 04:08:20,835 [INFO] (index.js) 202: Message received from player 1
09 Jun 2021 04:08:20,837 [INFO] (index.js) 210: Packet (including both internal messages and custom game messages) received from player: 1
09 Jun 2021 04:08:20,837 [INFO] (index.js) 380: Handling login request from player: 1
09 Jun 2021 04:08:20,837 [INFO] (index.js) 383: Player 1 logging in
09 Jun 2021 04:08:20,837 [INFO] (index.js) 390: Player: 1 is connecting to player session: psess-4357c5cf-fbd2-44b5-82f5-7209ce9ff77f
09 Jun 2021 04:08:20,838 [INFO] (gamelift.js) 90: Calling GameLiftServerAPI.AcceptPlayerSession with player session id: psess-4357c5cf-fbd2-44b5-82f5-7209ce9ff77f
09 Jun 2021 04:08:20,855 [INFO] (gamelift.js) 95: GameLiftServerAPI.AcceptPlayerSession succeeded with player session id: psess-4357c5cf-fbd2-44b5-82f5-7209ce9ff77f
09 Jun 2021 04:08:20,855 [INFO] (index.js) 397: Successfully connected player: 1 to player session:  psess-4357c5cf-fbd2-44b5-82f5-7209ce9ff77f
09 Jun 2021 04:08:20,855 [INFO] (index.js) 346: Received custom game message from sender: 1 with opCode: 113, targetPeerId: 1, targetGroupId: 0
09 Jun 2021 04:08:20,857 [INFO] (index.js) 432: Sending login result to player: 1
09 Jun 2021 04:08:21,143 [INFO] (server.js) 144: Tick... 1 activePlayers: 1
09 Jun 2021 04:08:22,145 [INFO] (server.js) 144: Tick... 2 activePlayers: 1`

Sample successful server log:
`08 Jun 2021 03:22:27,723 [INFO] (index.js) 471: Player 1 connected from <ip>
08 Jun 2021 03:22:27,723 [INFO] (ws.js) 139: Player 1 connecting from ::ffff:<ip>
08 Jun 2021 03:22:27,846 [INFO] (index.js) 202: Message received from player 1
08 Jun 2021 03:22:27,848 [INFO] (index.js) 210: Packet (including both internal messages and custom game messages) received from player: 1
08 Jun 2021 03:22:27,848 [INFO] (index.js) 380: Handling login request from player: 1
08 Jun 2021 03:22:27,848 [INFO] (index.js) 383: Player 1 logging in
08 Jun 2021 03:22:27,848 [INFO] (index.js) 390: Player: 1 is connecting to player session: psess-cddae252-4cf7-40fd-86f8-95df9b396898
08 Jun 2021 03:22:27,849 [INFO] (gamelift.js) 90: Calling GameLiftServerAPI.AcceptPlayerSession with player session id: psess-cddae252-4cf7-40fd-86f8-95df9b396898
08 Jun 2021 03:22:27,877 [INFO] (gamelift.js) 95: GameLiftServerAPI.AcceptPlayerSession succeeded with player session id: psess-cddae252-4cf7-40fd-86f8-95df9b396898
08 Jun 2021 03:22:27,878 [INFO] (index.js) 397: Successfully connected player: 1 to player session:  psess-cddae252-4cf7-40fd-86f8-95df9b396898
08 Jun 2021 03:22:27,878 [INFO] (index.js) 346: Received custom game message from sender: 1 with opCode: 113, targetPeerId: 1, targetGroupId: 0
08 Jun 2021 03:22:27,880 [INFO] (index.js) 432: Sending login result to player: 1
08 Jun 2021 03:22:27,949 [INFO] (index.js) 451: Sending server ack to player 1. iteration=0
08 Jun 2021 03:22:28,10 [INFO] (index.js) 282: Player 1 fully setup over ws + udp
08 Jun 2021 03:22:28,50 [INFO] (index.js) 442: Player 1 is now fully connected for tcp and udp
08 Jun 2021 03:22:28,158 [INFO] (server.js) 144: Tick... 1 activePlayers: 1
08 Jun 2021 03:22:29,159 [INFO] (server.js) 144: Tick... 2 activePlayers: 1`

**Comment by alanmur** [comment]
There was an updated release of the GameLift Realtime Servers SDK on June 3. Did you upgrade to the latest? It was intended to correct issues with IL2CPP. If you did, and that broke your game, that would be valuable information for us.

**Comment by hankditton** [comment]
Thanks for the follow up! This was not an IL2CPPP issue. 

I'm up and running now. I wish I could report back what I had to change, but I honestly am not sure. 

I reviewed the example client in the [MegaFrogRace ](https://github.com/aws-samples/megafrograce-gamelift-realtime-servers-sample/blob/master/MegaFrogRace/Assets/Scripts/RTSClient.cs) and made sure my connection code was making the exact same calls and now it's working. I don't have a history of what it was before or remember what I changed though. 

I do remember it was a very flight change that was not obvious at first. So if anyone else is having issues connecting, reset to that example and go from there. One difference I do remember was I was passing an empty byte[] payload in the ConnectionToken (instead of null), but I can't imagine that would have been the difference.

Apologies for the false alarm.
